src/state/CallViewModel.ts

```diff
@@ -1497,5 +1529,44 @@ export class CallViewModel extends ViewModel {
     >,
   ) {
     super();
+
+    // Pause all media tracks when we're disconnected from MatrixRTC, because it
+    // can be an unpleasant surprise for the app to say 'reconnecting' and yet
+    // still be transmitting your media to others.
+    this.matrixRTCConnected$.pipe(this.scope.bind()).subscribe((connected) => {
+      const publications =
+        this.livekitRoom.localParticipant.trackPublications.values();
+      if (connected) {
+        for (const p of publications) {
+          if (p.track?.isUpstreamPaused === true) {
```

Comment:
in the vm filter + forEach might fit better since it is more aligned with the observable syntax? wdyt? (`for` `if` also is nice because its so simple however...)

```suggestion
        publications
            .filter((p) => p.track?.isUpstreamPaused)
            .forEach((p) => {
            ...
            });
```

Reply:
It's a micro-optimization for sure, but I like avoiding the overhead of the filtered array in this case. I don't feel strongly about it though.

Reply:
lets keep it as is.